pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -5673,3 +5673,164 @@ func TestRhdmProdImmutableEnvironmentWithoutJbpmCluster(t *testing.T) {
+			Objects: api.KieAppObjects{
+				Servers: []api.KieServerSet{
+					{
+						Kafka: createKafkaExtObject(),
+					},
+				},
+			},
+		},
+	}
+
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+	assert.NotNil(t, env)
+	assert.True(t, len(cr.Spec.Objects.Servers[0].Kafka.Topics) == 2)
+	envs := env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env
+	autocreateEnv, maxBlockMS, topics, extEnabled, clientID, acks, groupID, bootstrapServers := false, false, false, false, false, false, false, false
+	for _, env := range envs {
+
+		switch e := env.Name; e {
+		case "KIE_SERVER_KAFKA_EXT_ENABLED":
+			extEnabled = true
+			assert.True(t, env.Value == "true")
+		case "KIE_SERVER_KAFKA_EXT_GROUP_ID":
+			groupID = true
+			assert.True(t, env.Value == "my-kafka-group")
+
+		case "KIE_SERVER_KAFKA_EXT_ACKS":
+			acks = true
+			assert.True(t, env.Value == "2")
+
+		case "KIE_SERVER_KAFKA_EXT_AUTOCREATE_TOPICS":
+			autocreateEnv = true
+			assert.True(t, env.Value == "true")
+
+		case "KIE_SERVER_KAFKA_EXT_MAX_BLOCK_MS":
+			maxBlockMS = true
+			assert.True(t, env.Value == "2100")
+
+		case "KIE_SERVER_KAFKA_EXT_CLIENT_ID":
+			clientID = true
+			assert.True(t, env.Value == "C1234567")
+
+		case "KIE_SERVER_KAFKA_EXT_SERVERS":
+			bootstrapServers = true
+			assert.True(t, env.Value == "localhost:9092")
+
+		case "KIE_SERVER_KAFKA_EXT_TOPICS":
+			topics = true
+			assert.True(t, env.Value == "events=my-topics,errors=my-errs")
+		}
+	}
+	assert.True(t, autocreateEnv)
+	assert.True(t, maxBlockMS)
+	assert.True(t, topics)
+	assert.True(t, extEnabled)
+	assert.True(t, clientID)
+	assert.True(t, acks)
+	assert.True(t, groupID)
+	assert.True(t, bootstrapServers)
+}
+
+func createKafkaExtObject() *api.KafkaExtObject {
+	kafkaExtObject := api.KafkaExtObject{
+		MaxBlockMs:       Pint32(2100),
+		AutocreateTopics: Pbool(true),
+		BootstrapServers: "localhost:9092",
+		GroupID:          "my-kafka-group",
+		Acks:             Pint(2),
+		Topics:           []string{"events=my-topics", "errors=my-errs"},
+		ExtEnabled:       true,
+		ClientID:         "C1234567",
+	}
+	return &kafkaExtObject
+}
+
+func TestRhdmEnvironmentWithKafkaExtDefault(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhdmProductionImmutable,
+			Objects: api.KieAppObjects{
+				Servers: []api.KieServerSet{
+					{
+						Kafka: &api.KafkaExtObject{},
+					},
+				},
+			},
+		},
+	}
+
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+	assert.NotNil(t, env)
+	envs := env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env
+
+	extEnabled := false
+	for _, env := range envs {
+		if strings.HasPrefix(env.Value, "KIE_SERVER_KAFKA") {
+			extEnabled = true
+		}
+	}
+	assert.False(t, extEnabled)
+
+	kafkaSpec := cr.Spec.Objects.Servers[0].Kafka
```

Comment:
this should be `kafkaSpec := cr.Status.Applied.Objects.Servers[0].Kafka`

we care about what's being applied. we already know what was set in the spec.

Reply:
IS present here: https://github.com/kiegroup/kie-cloud-operator/pull/614/files/6688118e35fe8ba23f66aeb7eb05c6ce10b6b2ba#diff-20d788a31791c7601d275bf50c8e30cb790834f4c05ce64c4e3bdfe92a6aaebaR5799